### DISCUSSION TITLE
TextInput allows wrapping on iOS instead of horizontal scroll or ellipsis

### DISCUSSION THREAD

**Discussion by winndana** [callstack/react-native-paper#3026] [Q&A]
When using the TextInput component, I have a problem where a long string of text entered will wrap to a second line, but only on iOS devices.  The expected behavior is a horizontal scroll or ellipsis.  I've tried setting nowrap, and I understand numberOfLines only applies to Android.  Is this a known issue?

**Comment by lukewalczak** [comment]
Hey @winndana, if I understand your issue correctly, you are facing the issue which is the [bug](https://github.com/facebook/react-native/issues/29068#issue-631523357) from react-native related to setting `textAlign` which we are handling within our [input](https://github.com/callstack/react-native-paper/blob/main/src/components/TextInput/TextInputFlat.tsx#L357) due to RTL.

More details: https://github.com/callstack/react-native-paper/issues/2581

Workaround: set `textAlign: 'auto'` in your input.
